### PR TITLE
improve(LM rewards): yUSD week 9 liquidity mining rewards

### PR DESCRIPTION
**Pull request by chrismaree**
**Overview**
This PR adds week 9's liquidity mining payouts. The scripts was run from from From from Sept 21, 11:00:00 PM UTC @ block 10908691 to sept 28, 11:00:00 PM UTC @ block 10953797. Over this period 25000 UMA were distributed.

**Details**
During this period there was a roll between two pools. During the roll, the yUSD provided to both pools was considered. There are three distinct phases considered in this roll, as shown in the diagram below.

1) `t_0->t_1`: "Old" pool from 21 Sept 22:00:00 UCT (10908691) to 23 Sept 22:00:00 UTC (10921632). 
- rewards to pay: (10921632 - 10908691) / (10953797 - 10908691) = 0.2869019643 of total rewards = 7173 UMA

2) `t_1->t_2`: Roll (considering both pools) from 23 Sept 22:00:00 (10921633) UTC until 27 Sept 22:00:00 (10947420)
- (10947420 - 10921633) / (10953797 - 10908691) = 0.5716977786 of total rewards = 14293 UMA

3) `t_2->t_3`: "New" pool from 27 Sept 22:00:00 UTC (10947421) until 28 22:00:00 UTC (10953797)
- (10953797 - 10947421) / (10953797 - 10908691) = 0.1413559172 of total rewards = 3534 UMA

Note that each period has an assigned number of UMA to distribute, summing to 25000 over the period (`7173 + 14293 + 3534 = 25000`).

**To re-create the payouts run the following**
1) Use `CalculateSplitPoolBalancerLPRewards.js` defined in #2028 which will generate the payouts for period `t_0` and `t_2` for the two normal pool periods. To run this execute:

```
node ./scripts/liquidity-mining/CalculateSplitPoolBalancerLPRewards.js --fromBlock1=10908691 --toBlock1=10921632 --fromBlock2=10947421 --toBlock2=10953797 --tokenName=yusdeth --pool1Address="0xd2f574637898526fcddfb3d487cc73c957fa0268" --pool2Address="0xcce41676a4624f4a1e33a787a59d6bf96e5067bc" --week=9 --umaPerPeriod1=7173 --umaPerPeriod2=3534
```
This generates payouts attributed to the two pools, allocating `7173` and `3534` UMA respectively. 

The outputs are placed in a `Week_9_Mining_Rewards.json` file which is similarly structured to previous payout files, except it persists each pool payouts, the block numbers associated with each and has the cumulative reward payout.

2) Run the `CalculateRollBalancerLPRewards.js` script to calculate the liquidity rewards, considering both pools:
```
node ./scripts/liquidity-mining/CalculateRollBalancerLPRewards.js --fromBlock 10921633 --toBlock 10947420 --pool1Address="0xd2f574637898526fcddfb3d487cc73c957fa0268" --pool2Address="0xcce41676a4624f4a1e33a787a59d6bf96e5067bc" --tokenName="yusdeth" --rollNum=2 --umaPerWeek=14293  --synth1Address="0xB2FdD60AD80ca7bA89B9BAb3b5336c2601C020b4" --synth2Address="0xD16c79c8A39D44B2F3eB45D2019cd6A42B03E2A9"
```
This generates payout considering the rolling liquidity over the two pools, allocating 14293 UMA to the roll. 

The output is placed in a `Roll_2_Mining_Rewards.json` file which contains the summated payouts.

3) Run the `JoinRolledPayouts.js` to combined the outputs from the `week 9` and `rollNum 2` scripts in steps 1 & 2.

```
node ./scripts/liquidity-mining/JoinRolledPayouts.js --week 9 --rollNum 2 --tokenName yusdeth
```

This creates an amended version of the `Week_9_Mining_Rewards.json` which contains the final payouts. the final `shareHolderPayout` is the total payouts for the weeks LM.

**Generated files**
The scripts creates two files in the output.
`Week_9_Mining_Rewards.json` contains the final payout amounts under `shareHolderPayout`. This object contains the joined payouts from the rolled period (two pools) and the normal farming period (one pool). The `weeklyShareHolderPayoutBeforeRollJoin` value in `Week_9_Mining_Rewards.json` is the weekly amounts from the normal (non-roll) period and is kept for prosperity and to enable auditability. The `Roll_2_Mining_Rewards.json` file contains the output during the roll. The `shareHolderPayout` in this file is the payouts for each shareholder during the roll period. Again, this is kept for auditability.

### CODE REVIEW

File: packages/core/scripts/liquidity-mining/yusdeth-weekly-payouts/Week_9_Mining_Rewards.json [hunk 1]
```diff
@@ -0,0 +1,986 @@
...
+    "0x8558f502887a9a52c4b265d72327e0e529ff790d": "2.943681798226849336",
+    "0x90e08dc4e7b9e87cb5c2558958488913fc3b3a00": "1.416398552351298138",
+    "0xb4a134bde4261a45bf60d19f7a51cc1c2c451523": "0.072507077048233621",
+    "0xb61d3430bd827b8c9a04e3f99896ccba674080da": "5.929943471993413426",
+    "0xd53c9fedcc95187307908d659846a443cb1e7350": "68.331098461047886598",
+    "0xdbfc163ae96a0e6f53e0296b9e03589871bbbf97": "6.388108241772963805",
+    "0xf75e32eea34ec4c326f340f26e9c7f0cde69cb02": "0.912614565499240597",
+    "0x4088e870e785320413288c605fd1bd6bd9d5bdae": "0.017774789600283191",
+    "0x5f4020aa88636cf382ac2a602e70e0831d18f54a": "5.582177789423377508",
+    "0x90162ad4623910bd53ea3c1f64acbf360678f8ba": "0.332397169126317949"
```
Comment (chrismaree) [inline packages/core/scripts/liquidity-mining/yusdeth-weekly-payouts/Week_9_Mining_Rewards.json:315]: These are the final weekly rewards.

File: packages/core/scripts/liquidity-mining/yusdeth-weekly-payouts/Week_9_Mining_Rewards.json [hunk 2]
```diff
@@ -0,0 +1,986 @@
...
+  "splitPeriodToBlock2": 10953797,
+  "tokenName": "yusdeth",
+  "splitPeriodPool1Address": "0xd2f574637898526fcddfb3d487cc73c957fa0268",
+  "splitPeriodPool2Address": "0xcce41676a4624f4a1e33a787a59d6bf96e5067bc",
+  "blocksPerSnapshot": 256,
+  "splitUmaPerPeriod1": 7173,
+  "splitUmaPerPeriod2": 3534,
+  "umaPerWeek": 25000,
+  "poolAddress": ["0xd2f574637898526fcddfb3d487cc73c957fa0268", "0xcce41676a4624f4a1e33a787a59d6bf96e5067bc"],
+  "shareHolderPayout": {
```
Comment (chrismaree) [inline packages/core/scripts/liquidity-mining/yusdeth-weekly-payouts/Week_9_Mining_Rewards.json:18]: this structure below is the final weekly payouts for week9

File: packages/core/scripts/liquidity-mining/yusdeth-weekly-payouts/Week_9_Mining_Rewards.json [hunk 3]
```diff
@@ -0,0 +1,986 @@
...
+    "0xb4a134bde4261a45bf60d19f7a51cc1c2c451523": "0.072507077048233621",
+    "0xb61d3430bd827b8c9a04e3f99896ccba674080da": "5.929943471993413426",
+    "0xd53c9fedcc95187307908d659846a443cb1e7350": "68.331098461047886598",
+    "0xdbfc163ae96a0e6f53e0296b9e03589871bbbf97": "6.388108241772963805",
+    "0xf75e32eea34ec4c326f340f26e9c7f0cde69cb02": "0.912614565499240597",
+    "0x4088e870e785320413288c605fd1bd6bd9d5bdae": "0.017774789600283191",
+    "0x5f4020aa88636cf382ac2a602e70e0831d18f54a": "5.582177789423377508",
+    "0x90162ad4623910bd53ea3c1f64acbf360678f8ba": "0.332397169126317949"
+  },
+  "splitPeriodShareHolderPayout1": {
```
Comment (chrismaree) [inline packages/core/scripts/liquidity-mining/yusdeth-weekly-payouts/Week_9_Mining_Rewards.json:321]: tokens allocated to the first few days in the "old" pool.

File: packages/core/scripts/liquidity-mining/yusdeth-weekly-payouts/Week_9_Mining_Rewards.json [hunk 4]
```diff
@@ -0,0 +1,986 @@
...
+    "0xf47c26277be32c4362d351badacae83f124e81e6": "0.431584442886924913",
+    "0xf4809688e2ac716baf617898243218da02773ee1": "0.759104034846423637",
+    "0xf53b2965d13404e5d13ce40c7448f8e13f04034b": "0.497239741680089818",
+    "0xf53cb49b4bc515c865e0503245621d797a9cc546": "16.66489593356792826",
+    "0xf94b3aa4415a14b289e329f9a75323656a2dcbc2": "2.220194735927131996",
+    "0xfdc6a17917115498a3ed417d4a89224685f26ea3": "8.427197375191671753",
+    "0xfdcfc1188005779ad1c70a0560fd5bbadaffebf5": "0.324316357558273531",
+    "0xffd44213718b48a371424303fe4aa2395ed4ce62": "886.729224052669121648"
+  },
+  "splitPeriodShareHolderPayout2": {
```
Comment (chrismaree) [inline packages/core/scripts/liquidity-mining/yusdeth-weekly-payouts/Week_9_Mining_Rewards.json:573]: tokens allocated to the last day in the "new" pool.

File: packages/core/scripts/liquidity-mining/yusdeth-weekly-payouts/Week_9_Mining_Rewards.json [hunk 5]
```diff
@@ -0,0 +1,986 @@
...
+    "0xef94af0495ddc90f0d81ad2487578e9c554c40e9": "8.905380640045133424",
+    "0xf01a613f03f977d6edd87eea3bd8d3355a177bcf": "1.701906278778031764",
+    "0xf1ee1f3c5b4fc684d50b61c458341a73abd63b1d": "0.130463626677005538",
+    "0xf4809688e2ac716baf617898243218da02773ee1": "0.545765276067424425",
+    "0xf737cf1a705c8ff6ac2592db084b8fe3c338daf5": "0.150432296843678194",
+    "0xf94b3aa4415a14b289e329f9a75323656a2dcbc2": "1.369219579961077497",
+    "0xfdc6a17917115498a3ed417d4a89224685f26ea3": "7.554155372465524422",
+    "0xffd44213718b48a371424303fe4aa2395ed4ce62": "501.811022279186687188"
+  },
+  "weeklyShareHolderPayoutBeforeRollJoin": {
```
Comment (chrismaree) [inline packages/core/scripts/liquidity-mining/yusdeth-weekly-payouts/Week_9_Mining_Rewards.json:698]: sum of the payouts between old pool and new pool, without the roll info.

File: packages/core/scripts/liquidity-mining/yusdeth-weekly-payouts/expiring-contract-rolls/Roll_2_Mining_Rewards.json [hunk 6]
```diff
@@ -0,0 +1,285 @@
+{
+  "rollNum": 2,
+  "fromBlock": 10921633,
+  "toBlock": 10947420,
+  "pool1Address": "0xd2f574637898526fcddfb3d487cc73c957fa0268",
+  "pool2Address": "0xcce41676a4624f4a1e33a787a59d6bf96e5067bc",
+  "umaPerWeek": 14293,
+  "blocksPerSnapshot": 256,
+  "shareHolderPayout": {
```
Comment (chrismaree) [inline packages/core/scripts/liquidity-mining/yusdeth-weekly-payouts/expiring-contract-rolls/Roll_2_Mining_Rewards.json:9]: rolled payout, considering both pools

Comment (chrismaree) [inline packages/core/scripts/liquidity-mining/yusdeth-weekly-payouts/expiring-contract-rolls/Roll_2_Mining_Rewards.json:9]: rolled payout, considering both pools.

File: packages/core/scripts/liquidity-mining/yusdeth-weekly-payouts/Week_9_Mining_Rewards.json [hunk 7]
```diff
@@ -0,0 +1,986 @@
...
+    "0x90e08dc4e7b9e87cb5c2558958488913fc3b3a00": "1.416398552351298138",
+    "0xb4a134bde4261a45bf60d19f7a51cc1c2c451523": "0.072507077048233621",
+    "0xb61d3430bd827b8c9a04e3f99896ccba674080da": "5.929943471993413426",
+    "0xd53c9fedcc95187307908d659846a443cb1e7350": "68.331098461047886598",
+    "0xdbfc163ae96a0e6f53e0296b9e03589871bbbf97": "6.388108241772963805",
+    "0xf75e32eea34ec4c326f340f26e9c7f0cde69cb02": "0.912614565499240597",
+    "0x4088e870e785320413288c605fd1bd6bd9d5bdae": "0.017774789600283191",
+    "0x5f4020aa88636cf382ac2a602e70e0831d18f54a": "5.582177789423377508",
+    "0x90162ad4623910bd53ea3c1f64acbf360678f8ba": "0.332397169126317949"
+  },
```
Comment (chrismaree) [inline packages/core/scripts/liquidity-mining/yusdeth-weekly-payouts/Week_9_Mining_Rewards.json:320]: These are the weekly rewards for yUSDwETH

**Review by nicholaspai — APPROVED** [review]

**Comment by hlambur** [comment]
Distributed with txs:
https://etherscan.io/tx/0x1c3cd9052cc8d9e6d86252f40dbd713b1381a64b8d43524f08695c06a531c6ed
https://etherscan.io/tx/0xa4063e300f53a0604281c0fc08e379772d9fc328adf24ab467b7e370c64472c9
https://etherscan.io/tx/0x9327c68a2e6d08e6698d54e2d7a3f38d46e84ed4464d638a90348f42cbaf9f4f

**Pull request merged**